### PR TITLE
Trev8939/deprecation fixes

### PR DESCRIPTION
**Pull request by TADraeseke**
Hi @eri9600 - This fixes use of highlighting width and color on feature layers and graphics overlays.

While in each of the relevant samples, I took the opportunity to fix some things:
- Use java 8 features like lambdas
- remove unecessary casting from findViewByIds
- Fix lifecycle ordering issues, etc

If you could cast an eye over this before I merge it'd be very helpful. Thank you!

### CODE REVIEW

File: java/offline-geocode/src/main/java/com/esri/arcgisruntime/sample/offlinegeocode/MainActivity.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public void run() {
 
     // add a graphics overlay
     graphicsOverlay = new GraphicsOverlay();
-    graphicsOverlay.setSelectionColor(Color.CYAN);
+    //graphicsOverlay.setSelectionColor(Color.CYAN);
```
Comment (elopezsolis) [inline java/offline-geocode/src/main/java/com/esri/arcgisruntime/sample/offlinegeocode/MainActivity.java]: You deleted this lines on the other samples, i don't know if you want to keep them comment out

File: java/edit-feature-attachments/src/main/java/com/esri/arcgisruntime/sample/editfeatureattachments/MainActivity.java [hunk 2]
```diff
@@ -271,8 +240,23 @@ protected void onResume() {
 
   @Override
   protected void onDestroy() {
-    super.onDestroy();
     mMapView.dispose();
+    super.onDestroy();
   }
 
+  /**
```
[the remark this answers is not in the document]
Comment (TADraeseke) [reply java/edit-feature-attachments/src/main/java/com/esri/arcgisruntime/sample/editfeatureattachments/MainActivity.java]: inner classes always get reordered to the bottom

File: java/feature-layer-update-geometry/src/main/java/com/esri/arcgisruntime/samples/featurelayerupdategeometry/MainActivity.java [hunk 3]
```diff
@@ -37,99 +37,100 @@
...
+                  Toast.makeText(getApplicationContext(), "Feature Selected. Tap on map to update its geometry.",
+                      Toast.LENGTH_LONG).show();
+                } else {
+                  Toast.makeText(getApplicationContext(), "No Features Selected. Tap on a feature.", Toast.LENGTH_LONG)
+                      .show();
                 }
-              } catch (InterruptedException | ExecutionException e) {
-                Log.e(getResources().getString(R.string.app_name), "Update feature failed: " + e.getMessage());
               }
+            } catch (InterruptedException | ExecutionException e12) {
```
[the remark this answers is not in the document]
Comment (TADraeseke) [reply java/feature-layer-update-geometry/src/main/java/com/esri/arcgisruntime/samples/featurelayerupdategeometry/MainActivity.java]: Haha, that one was very bizarre...good catch!

**Comment by TADraeseke** [comment]
Thanks @eri9600 - I'll keep an eye on Edit Feature Attachments (worked for me just now) and fix as required. Thanks for the review!

**Pull request merged**